include/envoy/router/router.h

```diff
@@ -603,6 +609,12 @@ class RouteEntry : public ResponseEntry {
    */
   virtual bool includeVirtualHostRateLimits() const PURE;
 
+  /**
+   * @return const Envoy::Config::TypedMetadata& return the typed metadata provided in the config
+   * for this route.
+   */
+  virtual const Envoy::Config::TypedMetadata& typedMetadata() const PURE;
```

Comment:
As with elsewhere, there should be no need for an `Envoy::` to namespaces for most Envoy code.